### ISSUE TITLE
Need to set DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER=true on linux builds 

### ISSUE THREAD

**Issue by MattGal** [dotnet/arcade#11369]
### Build

https://dev.azure.com/dnceng-public/public/_build/results?buildId=59411&view=results

### Build leg reported

Android_Simulators Build_Release

### Pull Request

https://github.com/dotnet/arcade/pull/11364

### Action required for the engineering services team

To triage this issue (First Responder / @dotnet/dnceng):
- [ ] Open the failing build above and investigate
- [ ] Add a comment explaining your findings 

If this is an issue that is causing build breaks across multiple builds and would get benefit from being listed on the [build analysis check](https://github.com/dotnet/arcade/blob/174acbcafeba3ae339c1935ec48812514ee09837/Documentation/Projects/Build%20Analysis/Introduction.md), follow the next steps:
1. Add the label "Known Build Error"
2. Edit this issue and add an error string in the Json below that can help us match this issue with future build breaks. You should use the [known issues documentation](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-message-section)

 ```json
 {
    "ErrorMessage" : "Unhandled exception: System.IO.IOException: Connection timed out : 'Global\\msbuild-server-launch-",
    "BuildRetry": false
 }
 ```

### Additional information about the issue reported

This is a known issue with .NET Core MSBuild in RC1 and RC2 and impacts any repo using this for build in linux 

### Report
|Build|Definition|Step Name|Console log|Pull Request|
|---|---|---|---|---|
|[63280](https://dev.azure.com/dnceng-public/public/_build/results?buildId=63280)|dotnet/roslyn|Build with analyzers|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/63280/logs/29)|dotnet/roslyn#64983|
|[63149](https://dev.azure.com/dnceng-public/public/_build/results?buildId=63149)|dotnet/roslyn|Build with analyzers|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/63149/logs/52)|dotnet/roslyn#64977|
|[62842](https://dev.azure.com/dnceng-public/public/_build/results?buildId=62842)|dotnet/roslyn|Build with analyzers|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/62842/logs/39)||
|[61673](https://dev.azure.com/dnceng-public/public/_build/results?buildId=61673)|dotnet/roslyn|Restore|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/61673/logs/31)||
|[61490](https://dev.azure.com/dnceng-public/public/_build/results?buildId=61490)|dotnet/roslyn|Restore|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/61490/logs/38)||
|[61357](https://dev.azure.com/dnceng-public/public/_build/results?buildId=61357)|dotnet/roslyn|Build with analyzers|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/61357/logs/45)|dotnet/roslyn#64936|
|[61126](https://dev.azure.com/dnceng-public/public/_build/results?buildId=61126)|dotnet/arcade|XHarness Android Helix Testing (Linux)|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/61126/logs/134)||
|[60694](https://dev.azure.com/dnceng-public/public/_build/results?buildId=60694)|dotnet/roslyn|Restore|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/60694/logs/36)||
|[59926](https://dev.azure.com/dnceng-public/public/_build/results?buildId=59926)|dotnet/roslyn|Restore|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/59926/logs/27)|dotnet/roslyn#64893|
|[59850](https://dev.azure.com/dnceng-public/public/_build/results?buildId=59850)|dotnet/arcade|Run Helix Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/59850/logs/107)|dotnet/arcade#11374|
|[59697](https://dev.azure.com/dnceng-public/public/_build/results?buildId=59697)|dotnet/roslyn|Restore|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/59697/logs/31)|dotnet/roslyn#64890|
|[59629](https://dev.azure.com/dnceng-public/public/_build/results?buildId=59629)|dotnet/arcade|XHarness Apple Simulator Helix Testing|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/59629/logs/134)||
|[59526](https://dev.azure.com/dnceng-public/public/_build/results?buildId=59526)|dotnet/roslyn|Restore|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/59526/logs/30)|dotnet/roslyn#64741|
|[59170](https://dev.azure.com/dnceng-public/public/_build/results?buildId=59170)|dotnet/arcade|XHarness Apple Simulator Helix Testing|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/59170/logs/106)|dotnet/arcade#11359|
|[59122](https://dev.azure.com/dnceng-public/public/_build/results?buildId=59122)|dotnet/arcade|XHarness Apple Simulator Helix Testing|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/59122/logs/105)|dotnet/arcade#11359|
|[58747](https://dev.azure.com/dnceng-public/public/_build/results?buildId=58747)|dotnet/roslyn|Restore|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/58747/logs/37)|dotnet/roslyn#64879|
|[58707](https://dev.azure.com/dnceng-public/public/_build/results?buildId=58707)|dotnet/roslyn|Restore|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/58707/logs/39)|dotnet/roslyn#64741|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|3|17|17|

**Comment by MattGal** [comment]
https://github.com/dotnet/arcade/pull/11371 should prevent this for arcade and anyone else using build.sh.  Roslyn does its own thing so not much we can do about that directly.

**Comment by MattGal** [comment]
@ilyas1974 I am tossing this into "tracking/blocked" since it should be fixed for arcade and users of tools.sh but Roslyn (top hitter) doesn't use this.  It should just go away at some point when newer .NET Core SDKs improve the functionality of the MSbuild server

**Comment by MattGal** [comment]
Discussed with @ilyas1974 ; since we have added the workaround in arcade and the only remaining hitter is using a custom build entry point, I am closing this issue as complete.  I reached out to some contacts on the Roslyn side to encourage workaround.

**Comment by dougbu** [comment]
@agocke mentioned in CI Council that server was disabled by default in 7.0.1xx SDK versions. Not sure of the exact version where that changed nor whether it's also in an available 8.0.1xx SDK. Is updating the SDK versions we use in release/7.0 and main a viable workaround now @rainersigwald @dsplaisted❔

If release/7.0 is still likely to hit this, suggest backporting the `DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER` to that Arcade branch (if you haven't already) @MattGal 

Separately, not this can impact integration and e2e testing on Helix agents. In aspnetcore, we added the environment variable in our test driver. But maybe it should be included in the Python code that does "stuff" for Helix work items❔

/cc @Chrisboh @ilyas1974 @steveisok @carlossanlop

**Comment by rainersigwald** [comment]
MSBuild Server is off by default in the GA build of 7.0.100 and 8.0 dailies, though there might have been a few 8.0 builds before that became true. If you're within the last week you should be fine. Updating to 7.0.100 GA should be viable after it releases but I don't think it's usable today because of packages.
